src/renderers/dom/client/wrappers/__tests__/DisabledInputUtil-test.js

```diff
@@ -105,4 +107,20 @@ describe('DisabledInputUtils', function() {
       });
     });
   });
+
+  describe('Object allocation', function () {
+    it('returns the same object if only given one object that is not disabled', function() {
+      var props = {};
+
+      expect(DisabledInputUtils.assignWithNativeProps(props)).toEqual(props);
+    });
+
+    it('non-destructively merges', function() {
+      var a = {};
+      var b = { value: true };
+      var props = DisabledInputUtils.assignWithNativeProps(a, b);
+
+      expect('value' in a).toEqual(false);
+    });
+  });
 });
```

Comment:
In case this gets touched again in the future, I went ahead and added some tests to ensure non-destructive behavior.